### PR TITLE
Add TextInputClient.currentAutofillScope

### PR DESCRIPTION
**Pull request by LongCatIsLooong**

### CODE REVIEW

File: src/docs/release/breaking-changes/add-currentAutofillScope-to-TextInputClient.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+On many platforms, autofill services are capable of autofilling multiple input fields in a single autofill 
+attempt. For example, username fields and password fields can usually be autofilled in one go. For this reason, 
+a Flutter input field that is about to trigger autofill should also provide the platform with information about other autofillable
+input fields logically connected to it. `TextInputClient.currentAutofillScope` defines the group of input fields that are logically
+connected to this `TextInputClient`, and can be autofilled together.
+
+## Description of change
+
+`TextInputClient` now has an additional getter that returns the `AutofillScope` that this client belongs to. This getter will be used by the input client to collect autofill related information from other
```
Comment (sfshaza2) [inline src/docs/release/breaking-changes/add-currentAutofillScope-to-TextInputClient.md:20]: Can you break the line at 80 chars or less?

Also:   will be => is

File: src/docs/release/breaking-changes/add-currentAutofillScope-to-TextInputClient.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+attempt. For example, username fields and password fields can usually be autofilled in one go. For this reason, 
+a Flutter input field that is about to trigger autofill should also provide the platform with information about other autofillable
+input fields logically connected to it. `TextInputClient.currentAutofillScope` defines the group of input fields that are logically
+connected to this `TextInputClient`, and can be autofilled together.
+
+## Description of change
+
+`TextInputClient` now has an additional getter that returns the `AutofillScope` that this client belongs to. This getter will be used by the input client to collect autofill related information from other 
+autofillable input fields within the same scope.
+
```
Comment (sfshaza2) [inline src/docs/release/breaking-changes/add-currentAutofillScope-to-TextInputClient.md:22]: You'll need a `<!-- skip -->` tag here.  Whenever you have code that is not a complete example, the analyzer fails so Travis fails. I can't merge a PR if Travis fails.

File: src/docs/release/breaking-changes/add-currentAutofillScope-to-TextInputClient.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+<!-- skip -->
+```dart
+class CustomTextFieldState extends State<CustomTextField> implements TextInputClient {
+  // Not having an AutofillScope does not prevent the input field
+  // from being autofilled. However, only this input field will
+  // be autofilled when autofill is triggered on it.
+  AutofillScope get currentAutofillScope => AutofillGroup.of(context);
+}
+```
+for more information about the `AutofillGroup` widget, please refer to its documentation: [`AutofillGroup`][].
```
Comment (sfshaza2) [inline src/docs/release/breaking-changes/add-currentAutofillScope-to-TextInputClient.md:54]: for => For

Also, delete "please". In fact, I'd amend this whole sentence to:

For more information, see [`AutofillGroup`][].

File: src/docs/release/breaking-changes/add-currentAutofillScope-to-TextInputClient.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+## Timeline
+
+TBA
+
+{% comment %}
+The version # of the SDK where this change was
+introduced.  If there is a deprecation window,
+the version # to which we guarantee to maintain the old API.
+{% endcomment %}
```
Comment (sfshaza2) [inline src/docs/release/breaking-changes/add-currentAutofillScope-to-TextInputClient.md:65]: Replace this comment with the version #, when you learn what that is. We don't add breaking changes until we know what release contains them.

File: src/docs/release/breaking-changes/add-currentAutofillScope-to-TextInputClient.md (hunk 2 above)
Comment (LongCatIsLooong) [reply src/docs/release/breaking-changes/add-currentAutofillScope-to-TextInputClient.md:27]: I expect the CI to pass once the proposed framework change lands. I'll add the tag if that doesn't fix this issue.

File: src/docs/release/breaking-changes/add-currentAutofillScope-to-TextInputClient.md [hunk 5]
```diff
@@ -0,0 +1,97 @@
+---
```
Comment (Hixie) [inline src/docs/release/breaking-changes/add-currentAutofillScope-to-TextInputClient.md:1]: don't forget to add this to the index as well

File: src/docs/release/breaking-changes/add-currentAutofillScope-to-TextInputClient.md [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+title: Add currentAutofillScope to TextInputClient
```
Comment (Hixie) [inline src/docs/release/breaking-changes/add-currentAutofillScope-to-TextInputClient.md:2]: This sounds like a PR title, not a migration guide title. Maybe "New currentAutofillScope property on TextInputClient"?

File: src/docs/release/breaking-changes/add-currentAutofillScope-to-TextInputClient.md [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+description: A new getter`TextInputClient.currentAutofillScope` was added to the `TextInputClient` interface for autofill support.
```
Comment (Hixie) [inline src/docs/release/breaking-changes/add-currentAutofillScope-to-TextInputClient.md:3]: missing space after `getter`

File: src/docs/release/breaking-changes/add-currentAutofillScope-to-TextInputClient.md [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+---
+
+## Summary
+
+A new getter`TextInputClient.currentAutofillScope` was added to the
```
Comment (Hixie) [inline src/docs/release/breaking-changes/add-currentAutofillScope-to-TextInputClient.md:8]: ditto.

File: src/docs/release/breaking-changes/add-currentAutofillScope-to-TextInputClient.md [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+If multifield autofill support is desirable, a common `AutofillScope` to use is
+the `AutofillGroup` widget. To get the closest `AutofillGroup` widget to the text
+input, use `AutofillGroup.of(context)`:
+
+<!-- skip -->
+```dart
+class CustomTextFieldState extends State<CustomTextField> implements TextInputClient {
+  // Not having an AutofillScope does not prevent the input field
+  // from being autofilled. However, only this input field will
+  // be autofilled when autofill is triggered on it.
```
Comment (Hixie) [inline src/docs/release/breaking-changes/add-currentAutofillScope-to-TextInputClient.md:57]: this comment is the same as in the previous snippet?

File: src/docs/release/breaking-changes/add-currentAutofillScope-to-TextInputClient.md [hunk 10]
```diff
@@ -0,0 +1,97 @@
+---
+title: Add currentAutofillScope to TextInputClient
+description: A new getter`TextInputClient.currentAutofillScope` was added to the `TextInputClient` interface for autofill support.
+---
+
+## Summary
+
+A new getter`TextInputClient.currentAutofillScope` was added to the
+`TextInputClient` interface.
```
Comment (Hixie) [inline src/docs/release/breaking-changes/add-currentAutofillScope-to-TextInputClient.md:9]: summary should include a little more detail than one sentence, maybe something like "Implementing this getting allows bla bla bla"

File: src/docs/release/breaking-changes/add-currentAutofillScope-to-TextInputClient.md [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+
```
Comment (Hixie) [inline src/docs/release/breaking-changes/add-currentAutofillScope-to-TextInputClient.md:10]: maybe also mention the context in which you might have your own TextInputClient subclass.

File: src/docs/release/breaking-changes/add-currentAutofillScope-to-TextInputClient.md (hunk 2 above)
Comment (sfshaza2) [reply src/docs/release/breaking-changes/add-currentAutofillScope-to-TextInputClient.md:33]: On the website repo, Travis tests the code against the current stable release. So I do expect the code to still fail. (The current stable release is 1.12.)

**Pull request merged**